BJS2-77457 | Система постов

Raidanmark opened the pull request:
Создание черновиков: с обязательным контентом и одним автором (пользователь или проект).
Публикация: с фиксацией даты, запрет повторной публикации.
Обновление: текста поста (автор неизменен).
Мягкое удаление.
Получение:
Поста по ID.
Черновиков (пользователя/проекта), сортировка по дате создания (новые сверху).
Опубликованных постов (пользователя/проекта), сортировка по дате публикации (новые сверху).

Raidanmark committed:
task created

AlexeyAntipin commented on src/main/java/faang/school/postservice/service/post/PostService.java:
```diff
@@ -0,0 +1,102 @@
+        postRepository.save(post);
+    }
+
+    @Transactional
```
readOnly = true

И везде, где транзакции только на чтение, нужно также

AlexeyAntipin commented on src/main/java/faang/school/postservice/service/project/ProjectService.java:
```diff
@@ -0,0 +1,30 @@
+            FeignException.GatewayTimeout.class},
+            backoff = @Backoff(delay = 1000, multiplier = 2)
+    )
+    public void checkProjectExist(Long userId) {
```
А зачем прослойка в виде сервиса? Почему сразу в нужное место не заинжектить клиент и не вызвать эндпоинт? И Retyrable лучше на метод клиента навесить

Raidanmark replied:
Ретраебл первесил, а касательно сервисной прослойки я просто думаю примерно так, все что касается к примеру проектов, валидация их, получение апдейт, короче все что угодно стоит делать в их сервисе, что бы если с ними пришлось какие то еще операции делать в другие, то просто заходить в этот сервис и смотретькакой метод юзать или еслиего нет дописывать. А если оно будет по всем расскидано то я могу одни и теже методы писать в разных сервисах снов и снова.

AlexeyAntipin commented on src/main/java/faang/school/postservice/service/user/UserService.java:
```diff
@@ -0,0 +1,32 @@
+                    FeignException.GatewayTimeout.class},
+            backoff = @Backoff(delay = 1000, multiplier = 2)
+    )
+    public void checkUserExist(Long userId) {
```
Такой же комментарий

AlexeyAntipin commented on src/main/java/faang/school/postservice/service/utils/PostServiceUtils.java:
```diff
@@ -0,0 +1,41 @@
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class PostServiceUtils {
```
Очень сложно читать, что тут происходит, попробуй как-то отрефакторить эту логику

Raidanmark committed:
Comment fixes

Raidanmark committed:
Comment fixes

Raidanmark committed:
Merge branch 'refs/heads/hydra-master-stream10' into BJS2-77457

Raidanmark committed:
Master merged

Raidanmark committed:
Refactor Jacoco configuration for better coverage control

Replaced exclusion-based filtering with inclusion-based rules for both report generation and coverage verification. Adjusted minimum coverage threshold from 70% to 80% to enforce stricter quality standards. Added dependency management between test, report, and verification tasks.

Raidanmark committed:
Merge branch 'decrease-procent-of-jacoco-coverage' into BJS2-77457

# Conflicts:
#	src/main/java/faang/school/postservice/dto/post/PostDto.java

Raidanmark committed:
Revert "Refactor Jacoco configuration for better coverage control"

This reverts commit 944a70024f0c140360a57300beb9708c18af32ca.

Raidanmark committed:
Revert "Master merged"

This reverts commit bcabf4f249e2def5e1a3b7cfdd4822177f97f79e.

Raidanmark committed:
Refactor PostService structure and update related utilities

Refactored PostService by incorporating PostServiceUtils for streamlined validation and retrieval logic, replacing redundant methods like isPostExists with getPost. Updated dependencies across services, tests, and controllers to adapt to these changes. Enhanced data validation by introducing DataValidationException for better error handling.
